packages/unmock-core/src/service/serviceStore.ts

```diff
@@ -1,91 +1,20 @@
-import { ISerializedRequest } from "../interfaces";
-import {
-  ExtendedHTTPMethod,
-  IService,
-  IServiceMapping,
-  isExtendedRESTMethod,
-  isStateInputGenerator,
-  IStateInputGenerator,
-  MatcherResponse,
-  UnmockServiceState,
-} from "./interfaces";
-import {
-  functionResponse,
-  objResponse,
-  textResponse,
-} from "./state/transformers";
+import { IService, IServiceCore, IServiceStore } from "./interfaces";
+import { Service } from "./service";
 
-export class ServiceStore {
-  private readonly serviceMapping: IServiceMapping = {};
+export const ServiceStore = (coreServices: IServiceCore[]) =>
+  new Proxy(
```

Comment:
Do we still need proxy if `unmock.services` is simply a `Record<string, Service> & ServiceStoreHelpers`?

Reply:
At least for me this type-checks fine:
```ts
interface Service {
  name: string;
}

type ServiceStoreHelpers = {
  findByHost(host: string): Service;
};

type Services = Record<string, Service> & ServiceStoreHelpers;

const service: Service = { name: "github " };

const services: Services = {
  findByHost(host: string) {
    if (host === "api.github.com") {
      return service;
    }
    throw Error("No service named github");
  },
  github: service,
};

const github = services.findByHost("api.github.com");
const githubAlt = services.github;
```

Reply:
Of course we may need custom getters to ensure that `services.githu` throws but I think the error message is quite obvious is one then tries to access `services.githu.state`.

Reply:
We don't need a proxy here at all, it's a question of do we want an error that reads `cannot read property 'state' of undefined` or `unmock: No service named '....'`.

:thinking: At least I'm not aware of how to throw when trying to access `services.githu`

Reply:
I prefer the `cannot read property 'state' of undefined` if it means avoiding proxy :)

Reply:
Sure. Personally, I think a more verbose message is beneficial for the end-user (just thinking how long it would take to notice if e.g. `services.gthub` is used by mistake, for example), but we can always revert back to this later on.

At this point then, `ServiceStore` has no functionality - it's just a mapping.

Reply:
`findByHost` left for another PR ofc